gapis/gfxapi/gfxapi.proto

```diff
@@ -102,6 +92,26 @@ enum UniformType {
 	Double = 4;
 }
 
+// ResourceData represents the resource state at a single point in a capture
+message ResourceData {
+	oneof data {
+		Texture texture = 1;
+		Shader shader = 2;
+		Program program = 3;
+	}
+}
+
+// Texture represents a texture resource.
+message Texture {
+	oneof type {
```

Comment:
> 3D, 2D_array has arbitrary number of layers,

Are you proposing that a 3D texture is just N x `image.Info2D`? If so, I think that's really clunky for the UI to use - especially if we need to actually recreate the 3D texture for a rotatable 3d view.

My personal opinion is that the server interface should be explicit - I steer away from structures that have fields (depth, layers) that only make sense when some other field (type) equals some magic value. This does result in more code, but should be clearer with less chance of misinterpretation.

> I have independently switched the GLSL API to follow the same model.

I'm more than happy for the internal model to be compact, my argument is just for the public interface.

This is purely my opinion though, so I'm happy to open this up for a vote.